clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -159,6 +159,9 @@ private void process(final PollEvent event) {
         } else {
             requestManagers.shareHeartbeatRequestManager.ifPresent(hrm -> hrm.resetPollTimer(event.pollTimeMs()));
         }
+
+        CompletableFuture<Void> future = requestManagers.fetchRequestManager.requestFetch();
```

Comment:
Hijacking the poll event like this is not quite going to work for the share consumer. In the share consumer, we use the PollEvent to signal that polling is occurring, and we have a separate ShareFetch event that asks the ShareConsumeRequestManager to fetch more records. There is no FetchRequestManager in a share consumer.

Reply:
I take umbrage at your use of "hijack." I was simply _borrowing_ it. 😏

Regardless, I've introduced a separate `FetchEvent` that should disentangle the two tasks.

Reply:
Thank you. Much nicer.